src/datachain/catalog/catalog.py

```diff
@@ -170,7 +174,7 @@ def check_for_status(self) -> None:
         Checks are done every PULL_DATASET_CHECK_STATUS_INTERVAL seconds
         """
         export_status_response = self.studio_client.dataset_export_status(
-            self.dataset_name, self.dataset_version
+            self.remote_ds_name, self.remote_ds_version
```

Comment:
[Q] Why not use `uuid` instead of `remote_ds_name` + `remote_ds_version` & `local_ds_name` + `local_ds_version`?

Reply:
Yea, there are other APIs that use `name` + `version` as well, but I would leave that to followup issue not to bloat this one too much. In general, user does use explicit name and version when initiating pull, e.g `datachain pull ds://dogs@v3` but the use of `uuid` in the process after initial get dataset info would fix potential issue of someone renaming remote dataset right in the middle of pull process (not sue how likely is this to happen but it's still the potential risk)

Reply:
None of this needs to go into this PR but I wanted to mention it and this seems like an ok place.

I would advocate for hiding local/remote name/version from users as much as possible. (From experience) I think the concept will only confuse users. Only exposing it when we absolutely need to makes sense to me 

For example:

If the user tries to pull `cat@v1` under the following circumstances we should throw an error stating there is a dataset with that name already and the uuid doesn't match. Ask them to rename the local dataset (as SaaS is the source of truth):

| name | version | uuid | local | remote |
| -- | -- | -- | -- | --| 
| cat | 1 | cf308a6b-0e66-477f-ba76-9c3f8ffd27fe | ✓ | X |
| cat | 1 | d1c1ac59-5763-4069-a328-8292ec6a6205 | X | ✓ |

I think we should update the output of `ls datasets` to have the same format as the above or at least give it the ability to show these discrepancies somehow.

I am now second-guessing myself and wondering whether the local/remote name and version are that bad.